validator/src/commands/plugin/mod.rs

```diff
@@ -54,41 +99,118 @@ pub fn execute(matches: &ArgMatches, ledger_path: &Path) -> Result<(), String> {
+                .map_err(|err| format!("unload plugin request failed: {err:?}"))?;
+            println!("Successfully unloaded plugin: {name}");
         }
         ("load", Some(subcommand_matches)) => {
-            if let Ok(config) = value_t!(subcommand_matches, "config", String) {
-                let admin_client = admin_rpc_service::connect(ledger_path);
-                let name = admin_rpc_service::runtime()
-                    .block_on(async { admin_client.await?.load_plugin(config.clone()).await })
-                    .map_err(|err| format!("load plugin request failed {config}: {err:?}"))?;
-                println!("Successfully loaded plugin: {name}");
-            }
+            let PluginLoadArgs { config } =
+                PluginLoadArgs::from_clap_arg_match(subcommand_matches)?;
+
+            let admin_client = admin_rpc_service::connect(ledger_path);
+            let name = admin_rpc_service::runtime()
+                .block_on(async { admin_client.await?.load_plugin(config.clone()).await })
+                .map_err(|err| format!("load plugin request failed {config}: {err:?}"))?;
+            println!("Successfully loaded plugin: {name}");
         }
         ("reload", Some(subcommand_matches)) => {
-            if let Ok(name) = value_t!(subcommand_matches, "name", String) {
-                if let Ok(config) = value_t!(subcommand_matches, "config", String) {
-                    let admin_client = admin_rpc_service::connect(ledger_path);
-                    admin_rpc_service::runtime()
-                        .block_on(async {
-                            admin_client
-                                .await?
-                                .reload_plugin(name.clone(), config.clone())
-                                .await
-                        })
-                        .map_err(|err| format!("reload plugin request failed {name}: {err:?}"))?;
-                    println!("Successfully reloaded plugin: {name}");
-                }
-            }
+            let PluginReloadArgs { name, config } =
+                PluginReloadArgs::from_clap_arg_match(subcommand_matches)?;
+
+            let admin_client = admin_rpc_service::connect(ledger_path);
+            admin_rpc_service::runtime()
+                .block_on(async {
+                    admin_client
+                        .await?
+                        .reload_plugin(name.clone(), config.clone())
+                        .await
+                })
+                .map_err(|err| format!("reload plugin request failed {name}: {err:?}"))?;
+            println!("Successfully reloaded plugin: {name}");
         }
         _ => unreachable!(),
     }
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn verify_args_struct_by_command_plugin_unload_default() {
+        let app = command();
+        let matches = app.get_matches_from_safe(vec![COMMAND, "unload"]);
+        assert!(matches.is_err());
```

Comment:
nit: Should we use the harness `verify_args_struct_by_command_is_error()` here ?